### DISCUSSION TITLE
Making Breeze a full blown Lucky app..

### DISCUSSION THREAD

**Discussion by jwoertink** [luckyframework/breeze#59] [Ideas]
While discussing some issues within Breeze, a suggestion was brought up that what if Breeze was just a full blown Lucky app? The way it would work is you'd boot it locally and set it up like a whole separate app. Then your app would send API requests to store all of the data.

### Benefits
* We would decouple Breeze from your app meaning less setup within your app
* Breeze routes and such would no longer be a part of your app
* Migrations would be separate and no longer an issue to your app. https://github.com/luckyframework/breeze/issues/47

### Downsides
* It would need a separate database setup
* Starting a new Lucky app that came with Breeze by default would require a few extra steps to make sure we know how to connect to your Breeze DB
* Breeze would need a fairly hefty rewrite as well as all of the integration stuff in the other shards with Pulsar

### Unknowns
* What happens if Breeze is using Lucky 0.28, but your app is on 0.29? Would they just not care?
* How do we run the breeze migrations from within your app? Or do you cd in to a different directory and run them like a separate app?

I'm not sold on this path, but I wanted to jot down the ideas for others to chime in. I'll update this as more thoughts come in.

**Comment by jkthorne** [comment]
Initially I was not sure about this but I think there a few good benefits here. I think another benefit is a benchmark app that can be referenced for examples and patterns.

Seems like some features like logging will be hard to do as a separate app.  This might have to be broken up into a shard and an app.

**Reply by jwoertink** [reply]
For sure. I have mixed feelings on it. I can see some benefits, but there's a lot of things that would make it really tricky.
